src/main/java/org/broadinstitute/hellbender/tools/spark/utils/LargeLongHopscotchSet.java

```diff
@@ -0,0 +1,224 @@
+            kryo.writeObject(stream, set);
+        }
+
+        kryo.setReferences(oldReferences);
+    }
+
+    public boolean add(final long entryValue) {
+        final int setIndex = setIndexOf(entryValue);
+        final LongHopscotchSet setRef = sets.get(setIndex);
+        return setRef.add(entryValue);
+    }
+
+    public void addAll(final long[] entryValues) {
+        for (final long val : entryValues) {
+            add(val);
+        }
+    }
+
+    public long size() {
+        long sum = 0;
+        for (final LongHopscotchSet s : sets) {
+            sum += s.size();
+        }
+        return sum;
+    }
+
+    public Collection<LongHopscotchSet> getSets() {
+        return sets;
+    }
+
+    public long capacity() {
+        long sum = 0;
+        for (final LongHopscotchSet s : sets) {
+            sum += s.capacity();
+        }
+        return sum;
+    }
+
+    public boolean contains(final long key) {
+        return sets.get(setIndexOf(key)).contains(key);
+    }
+
+    public boolean containsAll(final long[] vals) {
+        for (final long val : vals) {
+            if (!contains(val))
+                return false;
+        }
+        return true;
+    }
+
+    public LongIterator iterator() {
+        return new LargeLongHopscotchSetIterator();
+
+    }
+
+    public boolean remove(final long key) {
+        return sets.get(setIndexOf(key)).remove(key);
+    }
+
+    public boolean removeAll(final long[] set) {
+        boolean result = false;
+        for (final long val : set) {
+            if (remove(val)) result = true;
+        }
+        return result;
+    }
+
+    public boolean isEmpty() {
+        return this.size() == 0;
+    }
+
+    private int setIndexOf(final long val) {
+        return Integer.remainderUnsigned(longHash(val), numSets);
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        final LargeLongHopscotchSet that = (LargeLongHopscotchSet) o;
+
+        return numSets == that.numSets && sets.equals(that.sets);
+
+    }
+
+    @Override
+    public int hashCode() {
+        final LongIterator itr = iterator();
+        int sum = 0;
+        while (itr.hasNext()) {
+            sum += LongHopscotchSet.longHash(itr.next());
+        }
+        return sum;
+    }
+
+    public static final class Serializer extends com.esotericsoftware.kryo.Serializer<LargeLongHopscotchSet> {
+        @Override
+        public void write(final Kryo kryo, final Output output, final LargeLongHopscotchSet hopscotchSet) {
+            hopscotchSet.serialize(kryo, output);
+        }
+
+        @Override
+        public LargeLongHopscotchSet read(final Kryo kryo, final Input input, final Class<LargeLongHopscotchSet> klass) {
+            return new LargeLongHopscotchSet(kryo, input);
+        }
+    }
+
+    private final class LargeLongHopscotchSetIterator implements LongIterator {
```

Comment:
Just use an outer Iterator<LongHopscotchSet> and an inner LongIterator.
hasNext() checks the inner iterator for null or !hasNext and advances the outer iterator until it finds a new inner iterator that isn't empty, or until the outer iterator is exhausted.